clippy_lints/src/copy_then_borrow_mut.rs

```diff
@@ -0,0 +1,114 @@
+use clippy_config::Conf;
+use clippy_utils::diagnostics::span_lint_and_then;
+use clippy_utils::ty::is_copy;
+use clippy_utils::{get_enclosing_block, is_in_test, path_to_local};
+use rustc_ast::BindingMode;
+use rustc_errors::Applicability;
+use rustc_hir::{Block, BorrowKind, Expr, ExprKind, Mutability, Node, PatKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_session::impl_lint_pass;
+
+declare_clippy_lint! {
+    /// ### What it does
+    /// Checks for mutable reference on a freshly copied variable due to
+    /// the use of a block to return an value implementing `Copy`.
+    ///
+    /// ### Why is this bad?
+    /// Using a block will make a copy of the block result if its type
+    /// implements `Copy`. This might be an indication of a failed attempt
+    /// to borrow a variable, or part of a variable.
```

Comment:
```suggestion
    /// to borrow a variable.
```
I think we could word this differently to also specify field accesses but I don't think we need to, and how it's worded here just results in more confusion than it removes ("indication of part of a variable?", this is pedantic tho)

Reply:
Ok